lib/chef/provisioning/convergence_strategy/ignore_convergence_failure.rb

```diff
@@ -0,0 +1,53 @@
+class Chef
+module Provisioning
+  class ConvergenceStrategy
+
+    # The purpose of this class is to decore the `converge` method with logic to catch any
+    # convergence failure exceptions, log them and then squelch them.  The reason we
+    # need this is to prevent 1 provisioned node's converge failing an entire provisioning
+    # recipe.
+    module IgnoreConvergenceFailure
+
+      attr_accessor :ignore_failures_array, :ignore_exit_values
+
+      # This module is only meant to be extended into instances, not classes or modules
+      def self.extended(instance)
+        opts = instance.convergence_options[:ignore_failure]
+        instance.ignore_failures_array = []
+        instance.ignore_exit_values = []
+        if opts == true
+          instance.ignore_failures_array << RuntimeError
+        else
+          # We assume it is integers or errors
+          opts = [opts].flatten
+          opts.each do |o|
+            case
+            when o.is_a?(Fixnum)
+              instance.ignore_exit_values << o
+            when o.is_a?(Range)
+              instance.ignore_exit_values += o.to_a
+            when o <= Exception
+              instance.ignore_failures_array << o
```

Comment:
I dithered on that one.  Ultimately I decided to ignore unknown values so they don't get a runtime error halfway through a provisioning recipe because they specified an unknown option.  Also, I think this is non-intuitive enough that people won't be trying it randomly - they will be reading the `README` and examples.